src/ffmpegaudio.cc

```diff
@@ -52,11 +52,12 @@ AudioService::~AudioService()
 void AudioService::playMemory( const char * ptr, int size )
 {
   emit cancelPlaying( false );
+  if ( !thread.isNull() ) {
+    thread->wait();
+  }
   QByteArray audioData( ptr, size );
-  thread = std::make_shared< DecoderThread >( audioData, this );
-  connect( this, &AudioService::cancelPlaying, thread.get(), [ this ]( bool waitFinished ) {
-    thread->cancel( waitFinished );
-  } );
+  thread.reset( new DecoderThread( audioData, this ) );
```

Comment:
This `reset` here has the same meaning as the older `thread = std::make_shared` because the older code also ensure the existing one is deleted.